include/swift/Remote/MemoryReader.h

```diff
@@ -64,44 +64,20 @@ class MemoryReader {
   /// Attempts to read an integer of the specified size from the given
   /// address in the remote process.
   ///
-  /// Returns false if the operation failed, the request size is not
-  /// 1, 2, 4, or 8, or the request size is larger than the provided destination.
+  /// Returns false if the operation failed, or the request size is
+  /// larger than the provided destination.
   template <typename IntegerType>
   bool readInteger(RemoteAddress address, size_t bytes, IntegerType *dest) {
```

Comment:
A more flexible `readInteger` that can read integers with an arbitrary number of bytes, from 1 to `sizeof(*dest)`.  In particular, this is needed for handling 3-byte integers that can occur when storing enum tag indexes in the payload area.